server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java

```diff
@@ -266,37 +271,62 @@ private long removeTabletServerMarkers(Map<UUID,TServerInstance> uidMap,
     return result;
   }
 
-  private long removeFile(Path path) {
-    try {
-      if (!useTrash || !fs.moveToTrash(path)) {
-        fs.deleteRecursively(path);
+  private void removeFile(ExecutorService deleteThreadPool, Path path, AtomicLong counter,
+      String msg) {
+    deleteThreadPool.execute(() -> {
+      try {
+        log.debug(msg);
+        if (!useTrash || !fs.moveToTrash(path)) {
+          fs.deleteRecursively(path);
+        }
+        counter.incrementAndGet();
+      } catch (FileNotFoundException ex) {
+        // ignored
+      } catch (IOException ex) {
+        log.error("Unable to delete wal {}", path, ex);
       }
-      return 1;
-    } catch (FileNotFoundException ex) {
-      // ignored
-    } catch (IOException ex) {
-      log.error("Unable to delete wal {}", path, ex);
-    }
-
-    return 0;
+    });
   }
 
   private long removeFiles(Collection<Pair<WalState,Path>> collection, final GCStatus status) {
+
+    final ExecutorService deleteThreadPool = ThreadPools.getServerThreadPools()
+        .createExecutorService(context.getConfiguration(), Property.GC_DELETE_THREADS);
+    final AtomicLong counter = new AtomicLong();
+
     for (Pair<WalState,Path> stateFile : collection) {
       Path path = stateFile.getSecond();
-      log.debug("Removing {} WAL {}", stateFile.getFirst(), path);
-      status.currentLog.deleted += removeFile(path);
+      removeFile(deleteThreadPool, path, counter,
+          "Removing " + stateFile.getFirst() + " WAL " + path);
     }
-    return status.currentLog.deleted;
+
+    deleteThreadPool.shutdown();
+    try {
+      while (!deleteThreadPool.awaitTermination(1000, TimeUnit.MILLISECONDS)) { // empty
+      }
+    } catch (InterruptedException e1) {
+      log.error("{}", e1.getMessage(), e1);
+    }
+    return counter.get();
```

Comment:
The code was incrementing this deleted counter, should it still do that?

```suggestion
    status.currentLog.deleted += counter.get();
    return counter.get();
```

Reply:
Yep, I forgot to increment the counter. I created it though, so I was half way there.

Reply:
On second look, counter is being passed to removeFile, and incremented there.

Reply:
yeah counter is being incrementing removeFile, but the old code used to increment `status.currentLog.deleted` which may be used for display on the monitor and logging.  Seems like that is no longer incremented.

Reply:
Updated in 7c54495